### ISSUE TITLE
onnx2ncnn: MemoryData with input dims_size() == 4

### ISSUE THREAD

**Issue by xunings** [Tencent/ncnn#1247]
When converting a tf model to ncnn via onnx, I found the param of MemoryData was missing for the 4-dim input.
I understand tf is not officially supported, but it would be very helpful to raise at least a warning for this case, e.g., https://github.com/xunings/ncnn/commit/54ee7aac2171d216bcb0f7cc63de25d7922db37f

PS, in my case the 4-dim input was the alpha of the prelu layer. My tf model can be successfully converted to ncnn using the following workaround, 
https://github.com/xunings/ncnn/commit/9ebbececfe3fa094ecc689195ada3a4e2dc71dc6
But I am afraid this isn't a general solution.

**Comment by nihui** [comment]
We now recommend using keras2ncnn for converting tensorflow 1.x models, or mlir2ncnn for converting tensorflow 2.x models.

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
